### PR TITLE
amp-sticky-ad: add type check

### PR DESCRIPTION
**Pull request by zhouyx**

### CODE REVIEW

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 1]
```diff
@@ -23,6 +23,26 @@ import {toggle} from '../../../src/style';
 
 
 class AmpStickyAd extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
+    this.ad_ = element;
```
Comment (cramforce) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: Document why this is mutable.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 2]
```diff
@@ -23,6 +23,31 @@ import {toggle} from '../../../src/style';
...
+    /**
+     * This variable must not be nullable to be scheduleLayout later.
+     * It is mutable and temporarily assigned to element during construction.
+     * Will always change it during buildCallback.
+     * @private {!Element}
+     */
+    this.ad_ = element;
+
+    /** @const @private {!../../../src/service/viewport-impl.Viewport} */
+    this.viewport_ = this.getViewport();
```
Comment (erwinmombay) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: @dvoytenko would these servces be available before buildCallback?

Comment (dvoytenko) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: @erwinmombay These particular services - yes.

**Comment by zhouyx** [comment]
PTAL

**Comment by zhouyx** [comment]
Haven't submitted pending reviews from last Friday! LOL

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 3]
```diff
@@ -23,6 +23,26 @@ import {toggle} from '../../../src/style';
 
 
 class AmpStickyAd extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
```
Comment (cramforce) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: You want `scheduleLayout(dev().assertElement(this.ad_))` if you are sure that it is no longer `null`

Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: wait. I am confused. Do we want to modify the input type of `#scheduleLayout()` to accept nullable value?  Or do we want to assign a fake value to satisfy the `#scheduleLayout()` input must not be null check?   for example `this.ad_ = element` (element is actually the container element) 
I am applying the later one now. I believe https://github.com/ampproject/amphtml/blob/master/extensions/amp-sticky-ad/0.1/amp-sticky-ad.js#L36 already does the check for me

Comment (erwinmombay) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: no let's not change the signature of `scheduleLayout` but use `assertElement` on the call site if we can guarantee that you do have an element.

Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: Ha! I see what @cramforce means now. This is the perfect solution I am looking for! fixed. PTAL

**Pull request merged**